broadinstitute/single_cell_portal_core
# Improve error handling on gene search in a study (SCP-4373)

Previously if a user searched for a gene that didn't exist in a study there would be a 404 error thrown from rails but it was not being handled so it left the page with no results and a never-ending spinning loading icon.

https://user-images.githubusercontent.com/54322292/171872110-4b23522e-018e-4f90-9500-d011fb19c0a1.mov


Now if a user enters a gene that doesn't exist in the study first a modal will pop up to try to guide them to remove the offending gene from their search. However, if they persist and do the search anyway the error is handled and is shown to the user. Popping up the modal will reduce the failing network calls but even if they do it anyway the error will be made obvious.

![Screen Shot 2022-06-03 at 10 13 05 AM](https://user-images.githubusercontent.com/54322292/171884753-659af44d-94b3-481c-a560-1b59dbdeac25.png)


https://user-images.githubusercontent.com/54322292/171885553-56140668-73d3-4430-a814-c499b2adc68d.mov



To test:
- pull this branch
- open single cell portal and go to a study
- try searching for a variety of genes - in particular try misspelled ones, and combos of bad and good ones, plus try deleting and adding different genes.
- Observe in the console that there will be a 404 error but the error should be also displayed to the user in the plot region
